livekit-api/src/signal_client/mod.rs

```diff
@@ -111,14 +112,27 @@ impl Debug for SignalClient {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct SignalAnalyticsOptions {
+    pub sdk: String,
+    pub sdk_version: Option<String>,
+}
+
+impl Default for SignalAnalyticsOptions {
+    fn default() -> Self {
+        Self { sdk: "rust".to_string(), sdk_version: None }
+    }
+}
+
 impl SignalClient {
     pub async fn connect(
         url: &str,
         token: &str,
         options: SignalOptions,
+        analytics_options: SignalAnalyticsOptions,
```

Comment:
It'll break the API if we add this new arg

Reply:
let's put this inside of SignalOptions as a new optional field

Reply:
I had it that way originally - and happy to change it back - but FWIW it would break the API that way, too, depending on how you initialized your SignalOptions.

I think we should consider introducing this as a completely non breaking change with secondary versions of Room.connect and SignalClient.connect that take an extra argument to override these, and mark them as for internal use even though they're in the public SDK. it would minimize surface area of the change, while more cleanly delineating what these parameters are for.

Same basic thoughts in https://github.com/livekit/rust-sdks/pull/471#discussion_r1806794174, looking for feedback then I can finalize this PR @theomonnom @davidzhao

Reply:
For now I reverted the change made in https://github.com/livekit/rust-sdks/pull/471/commits/ce98197020136a37d6f44813b7cbe24af45392f6 so this is now back to the original proposal from an interface perspective:

FFI: The new parameters are required in `livekit_ffi_initialize`

Room: `Room::connect` is unchanged and a new `Room::connect_with_sdk` method is added to allow overriding the SDK. This keeps breaking changes out of the RTC package and avoids changes RoomOptions which are used in many places.

SignalClient: `SignalOptions` is extended to take sdk and sdk_version. This will be a breaking change for some users, depending on how they initialize the options, but within the API package is unlikely to cause issues as this project likely has very limited usage besides as a dependency of the RTC package.

Reply:
it's more unclear to ppl to know what connect_with_sdk is doing. I would vote to put that inside RoomOptions

Reply:
I think the right solution would have been to add #[non_exhaustive] to the Options struct in the earlier versions

Reply:
after further discussion with @theomonnom I've updated this to just include it in RoomOptions and SignalOptions. I chose to include it as a nested type rather than individual fields, as I anticipate adding at least one more field to it soon (for the other_sdks field to include the versions of associated sdks in use)

Reply:
Can you make the SignalOptions & RoomOptions #[non_exhaustive] instead?

Reply:
yeah sounds good - added that. i made the inner type non-exhaustive too so it won't be breaking later